### PR TITLE
cleanup claims

### PR DESCRIPTION
**Pull request by malaffoon**
I started coding the Target scoring and realized that Claims weren't in great shape.

### CODE REVIEW

File: data_generator/generators/iab_assessment.py [hunk 1]
```diff
@@ -79,35 +73,21 @@ def generate_interim_assessment(asmt_year: int,
...
-    sa.claim_1_score_weight = 1.0
-    sa.claim_2_score_min = 0
-    sa.claim_2_score_max = 0
-    sa.claim_2_score_weight = 0.0
-    sa.claim_3_score_min = 0
-    sa.claim_3_score_max = 0
-    sa.claim_3_score_weight = 0.0
-    sa.claim_4_score_min = None
-    sa.claim_4_score_max = None
-    sa.claim_4_score_weight = None
```
Comment (malaffoon) [inline data_generator/generators/iab_assessment.py]: The big change in this PR is un-flattening the claims so instead of 16 separate fields it is now a list of Claim or ClaimScore objects.

File: data_generator/generators/summative_or_ica_assessment.py [hunk 2]
```diff
@@ -194,41 +181,15 @@ def generate_assessment_outcome(date_taken: datetime.date,
...
-    sao.claim_2_score_range_max = min(assessment.claim_2_score_max, sao.claim_2_score + stderr)
-    sao.claim_2_perf_lvl = claim_perf_lvl(sao.claim_2_score, stderr, assessment.overall_cut_point_2)
-
-    sao.claim_3_score = claim_scores[2]
-    stderr = random_claim_error(sao.claim_3_score, assessment.overall_score_min, assessment.overall_score_max)
-    sao.claim_3_score_range_min = max(assessment.claim_3_score_min, sao.claim_3_score - stderr)
-    sao.claim_3_score_range_max = min(assessment.claim_3_score_max, sao.claim_3_score + stderr)
-    sao.claim_3_perf_lvl = claim_perf_lvl(sao.claim_3_score, stderr, assessment.overall_cut_point_2)
-
-    if assessment.claim_4_name is not None:
```
Comment (malaffoon) [inline data_generator/generators/summative_or_ica_assessment.py]: By making it a list, i don't have to worry that claim-4 isn't there for Math

File: data_generator/model/assessment.py [hunk 3]
```diff
@@ -91,3 +72,62 @@ def get_object_set(self):
         """
         return {'assessment': self,
                 'assessment_effective': {'date': write_filters.filter_date_Ymd(self.effective_date)}}
+
+    # These properties provide backward-compatible getters to claim info which was pushed into a list of objects
```
Comment (malaffoon) [inline data_generator/model/assessment.py]: This is a bit annoying: basically, the legacy output workers need to do a flat mapping of a field to a python object property, expecting None where appropriate. I couldn't do something like `asmt.claim[3].name` because `asmt.claim` might not be set or might have length 3 ... thus these wrapper properties.

Comment (malaffoon) [reply data_generator/model/assessment.py]: I guess i just need to bring myself to delete all the legacy output workers so i don't have to worry about stuff like this.

File: data_generator/util/assessment_stats.py [hunk 4]
```diff
@@ -225,9 +225,9 @@ def random_claims(score: int, claim_weights: [float], claim_min: int, claim_max:
         remaining_weight -= claim_weight
 
         min_ = min(claim_max,
-                   max(claim_min, math.floor((remaining_score - remaining_weight * claim_max) / claim_weight)))
+                   max(claim_min, int(math.floor((remaining_score - remaining_weight * claim_max) / claim_weight))))
         max_ = max(claim_min,
-                   min(claim_max, math.ceil((remaining_score - remaining_weight * claim_min) / claim_weight)))
+                   min(claim_max, int(math.ceil((remaining_score - remaining_weight * claim_min) / claim_weight))))
```
Comment (malaffoon) [inline data_generator/util/assessment_stats.py]: unrelated warning fix

File: test/generators/assessment_test.py [hunk 5]
```diff
@@ -49,42 +49,36 @@ def test_generate_assessment_invalid_subject():
...
     assert asmt.claim_2_score_max == 2769
-    assert asmt.claim_2_score_weight == .25
     assert asmt.claim_3_name == 'Listening'
     assert asmt.claim_3_score_min == 2288
     assert asmt.claim_3_score_max == 2769
-    assert asmt.claim_3_score_weight == .25
     assert asmt.claim_4_name == 'Research & Inquiry'
     assert asmt.claim_4_score_min == 2288
     assert asmt.claim_4_score_max == 2769
-    assert asmt.claim_4_score_weight == .3
```
Comment (malaffoon) [inline test/generators/assessment_test.py]: I removed `weight` from a claim because that is only needed when generating fake claim scores.

File: data_generator/generators/iab_assessment.py (hunk 1 above)
Comment (jtreuting) [reply data_generator/generators/iab_assessment.py]: does this mean that it could generate 6 levels of claims in the future for different subjects, or at least sounds like it's in a much better position to be able to do that ?

File: data_generator/model/assessment.py (hunk 3 above)
Comment (jtreuting) [reply data_generator/model/assessment.py]: seems like these legacy properties might be the only thing holding us back from generating 6 claim levels (though easy enough to add 2 more to the list i guess)

**Comment by agorina** [comment]
I am sure you are aware, but since you mentioned this change in relationship to Target, just want to clarify that the claims you have cleaned up are not the claims that are used for the Target Ids.

File: data_generator/generators/iab_assessment.py (hunk 1 above)
Comment (malaffoon) [reply data_generator/generators/iab_assessment.py]: yup

**Comment by malaffoon** [comment]
@agorina yes, these are the scorable claims.
